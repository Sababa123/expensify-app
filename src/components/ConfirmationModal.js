import React from 'react';
import Modal from 'react-modal';

const ConfirmationModal = (props) => (
    <Modal
        isOpen = {!!props.selected}
        contentLabel = "Selected Option"
        onRequestClose = {props.resetOnCheck}
        closeTimeoutMS = {200}
        className = "modal"
    >
        <h3 className="modal__title">Are you sure?</h3>
        <button className="button button--check" onClick={props.onRemove}>
        Yes</button>
        <button className="button button--check" onClick={props.resetOnCheck}>
        No</button>
    </Modal>
);

export default ConfirmationModal;
