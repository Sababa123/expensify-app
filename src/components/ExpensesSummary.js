import React from 'react';
import selectTotalExpenses from '../selectors/expenses-total';
import getVisibleExpenses from '../selectors/expenses';
import numeral from 'numeral';
import { connect } from 'react-redux';

export const ExpensesSummary = (props) => {
    const expenseWord = props.expenseCount === 1 ? 'expense' : 'expenses';
    const formattedTotal = numeral(props.expensesTotal/100).format('$0,0.00');
    return (
        <div>
            <h1>Viewing {props.expenseCount} {expenseWord} totalling {formattedTotal}</h1>
        </div>
    );
};

const mapStateToProps = (state,props) => {
    const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);
    return {
        expenseCount:  visibleExpenses.length,
        expensesTotal: selectTotalExpenses(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);