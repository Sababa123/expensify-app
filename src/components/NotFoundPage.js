import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
    <div>
        404! - <Link to="/">Go to</Link>
    </div>
);

export default NotFoundPage;