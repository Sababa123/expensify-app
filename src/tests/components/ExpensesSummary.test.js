import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import { shallow } from 'enzyme';

test('1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={50043}/>);
    expect(wrapper).toMatchSnapshot();
});

test('3 expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={1250043}/>);
    expect(wrapper).toMatchSnapshot();
});

