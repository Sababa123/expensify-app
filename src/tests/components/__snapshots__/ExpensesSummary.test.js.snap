// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $500.43
  </h1>
</div>
`;

exports[`3 expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $12,500.43
  </h1>
</div>
`;
